apps/files/lib/Controller/ConversionApiController.php

```
@@ -0,0 +1,95 @@
+		string $appName,
+		IRequest $request,
+		private IConversionManager $fileConversionManager,
+		private IRootFolder $rootFolder,
+		private IL10N $l10n,
+		private ?string $userId,
+	) {
+		parent::__construct($appName, $request);
+	}
+
+	/**
+	 * Converts a file from one MIME type to another
+	 *
+	 * @param int $fileId ID of the file to be converted
+	 * @param string $targetMimeType The MIME type to which you want to convert the file
+	 * @param string|null $destination The target path of the converted file. Written to a temporary file if left empty
+	 *
+	 * @return DataResponse<Http::STATUS_CREATED, array{path: string}, array{}>
+	 *
+	 * 201: File was converted and written to the destination or temporary file
+	 *
+	 * @throws OCSException The file was unable to be converted
+	 * @throws OCSNotFoundException The file to be converted was not found
+	 */
+	#[NoAdminRequired]
+	#[UserRateLimit(limit: 25, period: 120)]
+	#[ApiRoute(verb: 'POST', url: '/api/v1/convert')]
+	public function convert(int $fileId, string $targetMimeType, ?string $destination = null): DataResponse {
+		$userFolder = $this->rootFolder->getUserFolder($this->userId);
+		$file = $userFolder->getFirstNodeById($fileId);
+
+		if (!($file instanceof File)) {
+			throw new OCSNotFoundException($this->l10n->t('The file cannot be found'));
+		}
+
+		if ($destination !== null) {
+			$destination = PathHelper::normalizePath($destination);
+			$parentDir = dirname($destination);
+
+			if (!$userFolder->nodeExists($parentDir)) {
+				throw new OCSNotFoundException($this->l10n->t('The destination path does not exist: %1$s', [$parentDir]));
+			}
+
+			if (!$userFolder->get($parentDir)->isCreatable()) {
+				throw new OCSForbiddenException();
+			}
+
+			$destination = $userFolder->getFullPath($destination);
+		}
+
+		try {
+			$convertedFile = $this->fileConversionManager->convert($file, $targetMimeType, $destination);
+		} catch (\Exception $e) {
+			throw new OCSException($e->getMessage());
```

We should log the exception here and rather return a generic error message as the exception message may leak server internals. Also the trace is lost.